ci/infra/testrunner/skuba/skuba.py

```diff
@@ -45,27 +46,42 @@ def cleanup(conf):
         Utils.cleanup_files(dirs)
 
     @step
-    def cluster_init(self, kubernetes_version=None):
+    def cluster_init(self, kubernetes_version=None, cloud_provider=None):
         logger.debug("Cleaning up any previous test-cluster dir")
         self.utils.cleanup_file(self.cwd)
 
-        k8s_version_option = ""
+        k8s_version_option, cloud_provider_option = "", ""
         if kubernetes_version:
             k8s_version_option = "--kubernetes-version {}".format(kubernetes_version)
-        cmd = "cluster init --control-plane {} {} test-cluster".format(self.platform.get_lb_ipaddr(), k8s_version_option)
+        if cloud_provider:
+            cloud_provider_option = "--cloud-provider {}".format(cloud_provider)
```

Comment:
If you remove this from the `testrunner` cli you can retrieve the cloud provider from the `Skuba`  by adding this to the `__init__` method:

```
    def __init__(self, conf, platform):
          self.target_platform = platform
```
You can now use `self.target_platform`  as the name of the cloud provider.